scripts/gas-refund-program/transactions-indexing/swaps-subgraph.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  CHAIN_ID_AVALANCHE,
+  CHAIN_ID_BINANCE,
+  CHAIN_ID_FANTOM,
+  CHAIN_ID_MAINNET,
+  CHAIN_ID_POLYGON,
+} from '../../../src/lib/constants';
+import { Utils } from '../../../src/lib/utils';
+import { sliceCalls } from '../utils';
+
+const SwapsQuery = `
+query ($number_gte: BigInt, $number_lt: BigInt, $txOrgins: [Bytes!]!) {
+	swaps(
+		first: 1000
+		orderBy: blockNumber
+		orderDirection: asc
+		where: {
+			blockNumber_gte: $number_gte
+			blockNumber_lt: $number_lt
+			txOrigin_in: $txOrgins
```

Comment:
Are txOrigins always the accounts we are looking for? Even for SmartContract wallets like Argent and GnosisSafe?
What would the `initator` be for SmartContracts? The relayer?

Reply:
From observations, initiator could be smartcontracts where txOrigins is the top caller address. Checked  with samples including MetaMask aggregators